### DISCUSSION TITLE
Mock function with -OutVariable

### DISCUSSION THREAD

**Discussion by marais-vzc** [pester/Pester#2455] [General]
I have to mock the `Find-Package` cmdlet, but unable to because we use the `OutVariable` parameter. How would this be done?

I have done the following, but it doesn't work:

```
Mock Find-Package -ParameterFilter { $Name -eq "Pester" -and $OutVariable -eq "packages" -and [System.Boolean]$AllVersions -eq $AllVersions.IsPresent -and [System.Boolean]$AllowPrereleaseVersions -eq $AllowPrereleaseVersions.IsPresent -and $ErrorAction -eq "SilentlyContinue" } { return "SomePackage" }
```

**Comment by fflaten** [comment]
I assume you mean that the return value is not stored in the expected variable. I'm not aware of a good way to do this. You can replace `return "SomePackage"` with `Set-Variable -Scope 3 -Name $PesterBoundParameters.OutVariable -Value "SomePackage"`.

Be aware that the scope level `3` might need to be adjusted. It could also break in a future version of Pester as you're depending on how we wrap your code internally (not something we change often though). If `-Scope 3` doesn't work, try this:
1. Find (or create) a variable before the call to `Find-Package`
2. Add a breakpoint inside the Mock scriptblock (e.g. at `Set-Variable..`) and debug
3. When stopped at the breakpoint, run `Get-Variable -Name <varname> -Scope 2`, then 3, 4, etc. until it returns your variable's value. Use that number in `Set-Variable`.
